src/backend/Altinn.Receipt/Controllers/ReceiptController.cs

```diff
@@ -82,6 +87,14 @@ public async Task<IActionResult> GetCurrentUser()
             {
                 int userId = int.Parse(userIdString);
                 UserProfile profile = await _profile.GetUser(userId);
+
+                if (!string.IsNullOrEmpty(profile?.ProfileSettingPreference?.Language))
```

Comment:
This would use the cookie only if there are no profile or language preferences.

I was thinking that the cookie should take precedence above the profile settings. That way changes in the portal will take effect right away instead of needing to wait for the cache. 

This might introduce other issues though. I probably haven't considered all cases.

Reply:
Good point! I tried giving cookie the preference but it was throwing `NullReferenceException`. Now, of course by creating another endpoint dedicated for the language we can fix this, it need a bit of tweaking in the frontend which is a bit of extra work. 🤔 Anyway, will explore this path then. 🙂